pingcap/tidb-operator
# Don't update config of components when tls is enable

### What problem does this PR solve?

Close  #4298
### What is changed and how does it work?


### Code changes


### Tests



Please help test the following cases and make sure they are covered in e2e:

- Deploy TiDB cluster including PD, TiDB, TiKV, TiFlash, TiCDC, Pump
  - TLS not enabled
    - [x] Config for each component (TiFlash has two config fields) is `nil` (please note that some components like PD, TiDB, etc. cannot start with this config, just check the components that support this config)
    - [x] Config for each component (TiFlash has two config fields) is `{}`
    - [x] Config for each component (TiFlash has two config fields) has the valid config
  - TLS enabled
    - [x] Config for each component (TiFlash has two config fields) is `nil` (please note that some components like PD, TiDB, etc. cannot start with this config, just check the components that support this config)
    - [x] Config for each component (TiFlash has two config fields) is `{}`
    - [x] Config for each component (TiFlash has two config fields) has the valid config
- Deploy DM cluster including DM Master and DM worker (Same cases for TiDB cluster)
  - [x] TLS not enabled
  - [x] TLS enabled


### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```
